v0/destinations/yahoo_dsp/transform.js

```diff
@@ -0,0 +1,201 @@
+
+const {
+  getAccessToken,
+  populateIncludes,
+  populateExcludes,
+  createPayload
+} = require("./util");
+
+/**
+ * This function is used for building the final response to be returned.
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+
+const responseBuilder = async (message, destination) => {
+  const { listData } = message.properties;
+  const {
+    accountId,
+    audienceId,
+    audienceType,
+    seedListType
+  } = destination.Config;
+
+  let outputPayload = {};
+  // The only supported property for now is "add"
+  const key = "add";
+
+  const domains = [];
+  const categoryIds = [];
+  const listDataKey = listData[key];
+  if (!listDataKey) {
+    throw new CustomError(
+      `[Yahoo_DSP]:: The only supported operation for audience updation '${key}' is not present`,
+      400
+    );
+  }
+
+  /**
+   * The below written switch case is used to build the response for each of the supported audience type.
+   *  eg. ["email", "deviceId", "ipAddress", "mailDomain", "pointOfInterest"].
+   */
+  switch (audienceType) {
+    case "email":
+      // creating the output payload using the audience list and Config
+      outputPayload = createPayload(listDataKey, destination.Config);
+      break;
+    case "deviceId":
+      // throwing error if seedListType is not provided for deviceId type audience
+      if (
+        !seedListType ||
+        (seedListType && seedListType !== "IDFA" && seedListType !== "GPADVID")
+      ) {
+        throw new CustomError(
+          `[Yahoo_DSP]:: seedListType is required for deviceId type audience and it should be any one of 'IDFA' and 'GPADVID'`,
+          400
+        );
+      }
+      outputPayload = createPayload(listDataKey, destination.Config);
+      // updatig seedListType here
+      outputPayload = {
+        ...outputPayload,
+        seedListType
+      };
+      break;
+    case "ipAddress":
+      outputPayload = createPayload(listDataKey, destination.Config);
+      // updating seedListType in outputPayload
+      outputPayload = {
+        ...outputPayload,
+        seedListType: "SHA256IP"
+      };
+      break;
+    case "mailDomain":
+      // traversing through every element in the add array for the elements to be added.
+      listDataKey.forEach(element => {
+        // storing keys of an object inside the add array.
+        const keys = Object.keys(element);
+        // For mailDomain the mailDomain or categoryIds must be present. throwing error if not present.
+        /**
+         * If mailDomain is not provided categoryIds is required. The audience includes consumers who have received
+         * mail from a domain belonging to the specified categories.
+         * Reference for use case of categoryIds:
+         * https://developer.yahooinc.com/dsp/api/docs/traffic/audience/mrt-audience.html#:~:text=Optional-,categoryIds,-Specifies%20an%20array
+         */
+        if (!(keys.includes(audienceType) || keys.includes("categoryIds"))) {
```

Comment:
Please use enum for `audienceType` also for checking `categoryIds` use constant from config.js

Reply:
done